roboflow/models/object_detection.py

```diff
@@ -132,27 +133,42 @@ def predict(
             stroke=stroke,
             labels=labels,
         )
-        # Check if image exists at specified path or URL
-        self.__exception_check(image_path_check=image_path)
+
+        # Check if image exists at specified path or URL or is an array
+        if hasattr(image_path, "__len__") == True:
+            pass
+        else:
+            self.__exception_check(image_path_check=image_path)
 
         # If image is local image
         if not hosted:
-            # Open Image in RGB Format
-            image = Image.open(image_path).convert("RGB")
-
-            # Create buffer
-            buffered = io.BytesIO()
-            image.save(buffered, quality=90, format="JPEG")
-            # Base64 encode image
-            img_str = base64.b64encode(buffered.getvalue())
-            img_str = img_str.decode("ascii")
-
-            # Post to API and return response
-            resp = requests.post(
-                self.api_url,
-                data=img_str,
-                headers={"Content-Type": "application/x-www-form-urlencoded"},
-            )
+            if ".jpg" in image_path or ".png" in image_path:  # Open Image in RGB Format
+                image = Image.open(image_path).convert("RGB")
+
+                # Create buffer
+                buffered = io.BytesIO()
+                image.save(buffered, quality=90, format="JPEG")
```

Comment:
why JPEG and 90 quality?

Reply:
discussed here as well -> https://github.com/roboflow-ai/roboflow-python/pull/68 (in the comments) 👀

Reply:
I removed the quality flag and changed the format to "PNG"

It was legacy from another script.